app/controllers/vm_common.rb

```diff
@@ -928,6 +928,14 @@ def parent_choices_with_no_parent_choice
     {'"no parent"' => -1}.merge(parent_choices(@record.id))
   end
 
+  # Return vm_cloud or vm_infra based on selected record
+  def vm_or_instance
+    if @record || @tagitems
```

Comment:
Why the `@tagitems` what's the case for that?

Reply:
@tagitems contains items we selected for policy simulation. I did not created this, I am just using it. @record is not always set - it depends: explorer vs. non explorer screens.

Reply:
So is `@tagitems` always set? Why do you need both variables? What's the difference between the two behavior?

Reply:
Once the @record is set and if not, then @tagitems is set for sure. Yes, I need bot variables. It depends from where you use this method, and also it depends whether you go to policy simulation screen thru nested list of items (non explorer) or not (explorer screen) or from summary page of some item (VM/Instance...). It all works differently for each of these situations, different variables are set, different code. Hope this helps.